### DISCUSSION TITLE
[RESOLVED] ACD (The ansible package)

### DISCUSSION THREAD

**Discussion by gundalow** [ansible-collections/overview#39] [Proposals] [answered]
# SUMMARY

# Use-cases

## Users

- As a user I want to 
## Contributors

- As a Contributor I 
## Maintainers

- As a Maintainer I 
## Community Team

- As a member of the Community Team

# ACD Package

## Build system

# Outstanding questions
(Please don't delete numbers as this will make reviewing comments difficult)
1.

**Comment by geerlingguy** [comment]
One concern I have now that I'm in the very preliminary stages of figuring out how I might migrate some of my roles into roles-in-collections: because it looks like using an unmodified role that calls modules that are not called using a FQCN is [not going to be supported](https://github.com/ansible/ansible/issues/68198), it may be challenging (or at least something that requires more work and testing) to make sure that a role which might maintain compatibility with Ansible 2.8, 2.9, ACD, _and_ 'Ansible whatever' (other non-ACD Ansible ditributions) (intentionally leaving out 2.7 and earlier...) will work in all these scenarios.

Because of the namespacing (and ACD maybe not requiring that namespacing?), I might have to decide if my role works on ACD and not-other-Ansible-distributions, or vice-versa. And even if normal _playbooks_ work under ACD, is that a guarantee that _roles_ will also work (it was a huge surprise to me to see that my playbook role—_and I'm not talking about Galaxy roles, just roles local to a playbook in general_—did not work the same whether tasks in the role were in the role or in the playbook proper.

I had assumed that roles would work like they always have, where I can drop a task from a role into a playbook, and vice-versa, and that task would behave the same.

In the new world order, because of this new behavior where tasks in roles don't inherit the same functionality as tasks in playbooks, I can imagine people getting extremely confused about why roles don't work the same between various flavors of Ansible.

**Comment by gregdek** [comment]
Discussion is in the following gist:

https://gist.github.com/abadger/1f14caded92117fbd3036842c875701c

**Comment by gundalow** [comment, marked as answer]
Resolved
